src/kemal/config.cr

```diff
@@ -49,22 +49,21 @@ module Kemal
     def setup_logging
       if @logging
         @logger ||= Kemal::CommonLogHandler.new(@env)
-        HANDLERS << @logger.not_nil!
+        HANDLERS.insert(0, @logger.not_nil!)
       else
         @logger = Kemal::NullLogHandler.new(@env)
-        HANDLERS << @logger.not_nil!
```

Comment:
Because it's not needed, it will still be callable, but it doesn't need to be part of the middleware stack since it does nothing.